### ISSUE TITLE
[ResourceBundle] EntityRepository assumes IDs are integers

### ISSUE THREAD

**Issue by LewisW** [Sylius/Sylius#3379]
[Calling find in the EntityRepository assumes entity IDs are an integer](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ResourceBundle/Doctrine/ORM/EntityRepository.php#L47):

```
    public function find($id)
    {
        return $this
            ->getQueryBuilder()
            ->andWhere($this->getAlias().'.id = '.intval($id))
            ->getQuery()
            ->getOneOrNullResult()
        ;
    }
```

Which is not the case if you're using a UUID. It also hard-codes the primary key as being called `id`. I was going to submit a PR that fixes this, but I can't see any additional functionality the the `find` method provides over Doctrine's entity repository? The base entity repository already handles finding the primary key's column name and escaping inputs, so shouldn't we just rely on that instead?

**Comment by michalmarcinkowski** [comment]
Will be fixed after https://github.com/Sylius/Sylius/pull/2255 merge

**Comment by LewisW** [comment]
Brilliant, thanks!

**Comment by michalmarcinkowski** [comment]
Fixed in #3854.
